comp/core/workloadmeta/collectors/internal/remote/generic.go

```diff
@@ -58,7 +58,7 @@ type StreamHandler interface {
 	// NewClient returns a client to connect to a remote gRPC server.
 	NewClient(cc grpc.ClientConnInterface) GrpcClient
 	// HandleResponse handles a response from the remote gRPC server.
-	HandleResponse(response interface{}) ([]workloadmeta.CollectorEvent, error)
+	HandleResponse(store workloadmeta.Component, response interface{}) ([]workloadmeta.CollectorEvent, error)
```

Comment:
since `HandleResponse` function is part of GenericCollector` that already get access to the `store` we it is started in the `Start()` function. why do we need to path it as argument here?